quickwit/quickwit-serve/src/index_api/rest_handler.rs

```diff
@@ -47,75 +51,160 @@ pub fn index_management_handlers(
 )]
-/// Get All Splits
-async fn get_all_splits(
+/// Get splits.
+async fn get_splits(
     index_id: String,
-    index_service: Arc<IndexService>,
-) -> Result<Vec<Split>, IndexServiceError> {
-    info!(index_id = %index_id, "get-all-splits");
-    index_service.get_all_splits(&index_id).await
+    list_split_query: ListSplitsQueryParam,
+    metastore: Arc<dyn Metastore>,
+) -> Result<Vec<Split>, MetastoreError> {
+    info!(index_id = %index_id, list_split_query = ?list_split_query, "get-splits");
+    let mut query = ListSplitsQuery::for_index(&index_id);
+    if let Some(split_states) = list_split_query.split_states {
+        query = query.with_split_states(split_states);
+    }
+    if let Some(start_timestamp) = list_split_query.start_timestamp {
+        query = query.with_time_range_start_gte(start_timestamp);
+    }
+    if let Some(end_timestamp) = list_split_query.end_timestamp {
+        query = query.with_time_range_end_lt(end_timestamp);
+    }
+    if let Some(end_created_timestamp) = list_split_query.end_create_timestamp {
+        query = query.with_create_timestamp_lt(end_created_timestamp);
+    }
+    metastore.list_splits(query).await
 }
 
-fn get_all_splits_handler(
-    index_service: Arc<IndexService>,
+fn get_splits_handler(
+    metastore: Arc<dyn Metastore>,
 ) -> impl Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
     warp::path!("indexes" / String / "splits")
         .and(warp::get())
-        .and(with_arg(index_service))
-        .then(get_all_splits)
+        .and(serde_qs::warp::query(serde_qs::Config::default()))
+        .and(with_arg(metastore))
+        .then(get_splits)
+        .map(format_response)
+}
+
+#[derive(Deserialize)]
+#[serde(deny_unknown_fields)]
+struct SplitsForDeletion {
+    pub split_ids: Vec<String>,
+}
+
+#[utoipa::path(
+    post,
+    tag = "Splits",
+    path = "indexes/{index_id}/splits/mark-splits-for-deletion",
+    request_body = SplitsForDeletion,
+    responses(
+        (status = 200, description = "Successfully marked splits for deletion.")
+    ),
+    params(
+        ("index_id" = String, Path, description = "The index ID to mark splits for deletion for."),
+    )
+)]
+/// Mark splits for deletion.
+async fn mark_splits_for_deletion(
+    index_id: String,
+    splits_for_deletion: SplitsForDeletion,
+    metastore: Arc<dyn Metastore>,
+) -> Result<(), MetastoreError> {
+    info!(index_id = %index_id, splits_ids = ?splits_for_deletion.split_ids, "mark-splits-for-deletion");
+    let split_ids: Vec<&str> = splits_for_deletion
+        .split_ids
+        .iter()
+        .map(|split_id| split_id.as_ref())
+        .collect();
+    metastore
+        .mark_splits_for_deletion(&index_id, &split_ids)
+        .await
+}
+
+fn mark_splits_for_deletion_handler(
+    metastore: Arc<dyn Metastore>,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
+    warp::path!("indexes" / String / "splits" / "mark-splits-for-deletion")
```

Comment:
Should we switch to DELETE instead of POST?
Also making me think that the endpoint could be named `indexes/{index_id}/slipts/delete` because the `mark-splits-for-deletion` is an internal implementation detail. I would be explicit about this if there were a way to recover (undo) the marked split.

Reply:
Good point; I can shorten it to `mark-for-deletion`. I like to remain explicit, and we only change the state for now (if the GC is not running or has some delays), this will be visible anyway. 
Also, consequently, I would keep the `POST` method instead of `DELETE`.

Reply:
We can use `PUT`instead of `POST`.